ginga/Control.py

```diff
@@ -517,29 +514,37 @@ def load_image(self, filepath, idx=None):
             filepath = filepfx
         else:
             inherit_prihdr = self.settings.get('inherit_primary_header', False)
-            image = AstroImage.AstroImage(logger=self.logger,
-                                          inherit_primary_header=inherit_prihdr)
+            image = AstroImage.AstroImage(
+                logger=self.logger, inherit_primary_header=inherit_prihdr)
             kwdargs.update(dict(numhdu=idx))
 
         try:
             self.logger.info("Loading image from %s kwdargs=%s" % (
                 filepath, str(kwdargs)))
             image.load_file(filepath, **kwdargs)
 
-        except Exception as e:
-            errmsg = "Failed to load file '%s': %s" % (
-                filepath, str(e))
-            self.logger.error(errmsg)
+        except Exception:
+            # TODO: Refactor code to be less hacky.
+            # Maybe it has a table on EXT 1.
             try:
```

Comment:
Probably we should add a new method to the fits loader classes in `io_fits` that looks a bit like `load_file`, but instead of returning a tuple with the data (and filling in a passed header) it creates the right kind of `AstroImage` or `AstroTable` and then calls the right methods to load it and return it.